packages/astroport-governance/src/utils.rs

```diff
@@ -7,8 +7,12 @@ pub const MAX_LOCK_TIME: u64 = 2 * 365 * 86400; // 2 years (104 weeks)
 /// The constant describes the maximum number of accounts for claim.
 pub const CLAIM_LIMIT: u64 = 10;
 
+const PERIOD_SHIFT: u64 = 4 * 86400;
+
 /// # Description
 /// Calculates how many periods are within specified time. Time should be in seconds.
 pub fn get_period(time: u64) -> u64 {
-    time / WEEK
+    // Timestamp = 0 starts on Thursday, 1 January 1970
+    // So we need to shift it by 4 days to start our periods on Monday
+    (time + PERIOD_SHIFT) / WEEK
```

Comment:
For example: on Monday, the distributor received 100 commissions, with a PERIOD_SHIFT they will be recorded for the current period. The user makes a lock on a Friday this week, but with a PERIOD_SHIFT the lock will be created next week, Tuesday. As a result, the user will lose their rewards from the distributor this week.

Reply:
The issue you are talking about exists right now in the current code. This PR chage will fix this issue.
I think this can help to understand.

```bash
mon_ts=$(date -d 'Feb 28 2022 07:21:22' +%s)
fri_ts=$(date -d 'Mar 4 2022 07:21:22' +%s)

echo "Wrong:"
echo $((mon_ts/(86400*7)))
echo $((fri_ts/(86400*7)))

echo "Valid:"
echo $(((fri_ts+86400*4)/(86400*7)))
echo $(((mon_ts+86400*4)/(86400*7)))
```

Reply:
For Sunday this work only to 01:59:59, after we will get wrong period:

sun_ts=$(date -d 'Mar 06 2022 01:59:59' +%s)
Valid: 2722

sun_ts=$(date -d 'Mar 06 2022 02:00:00' +%s)
Wrong: 2723

sun_ts=$(date -d 'Mar 06 2022 23:59:00' +%s)
Wrong: 2723

Reply:
Thanks, fixed to shift time by 3 days.